### PR TITLE
Add UI tests for speedgrader comments

### PR DESCRIPTION
**Pull request by cobbal**
refs: mbl-14014
affects: none
release note: none

[run-nightly]

### CODE REVIEW

File: rn/Teacher/src/canvas-api/apis/media-comments.js [hunk 1]
```diff
@@ -133,7 +133,8 @@ async function completeUpload (id, type, context, contextID) {
 // UTILS
 
 function formatDomain (domain) {
-  if (!domain.startsWith('https')) {
+  // localhost domain used for testing over an unsecured connection
+  if (!domain.startsWith('https') && !domain.startsWith('http://localhost:')) {
```
Comment (cobbal) [inline rn/Teacher/src/canvas-api/apis/media-comments.js:137]: Not the ideal solution, but I'm reasonably certain it won't break anything

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/dbepr52) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fdbepr52)
teacher: [Link](https://rebrand.ly/4bmkrq3) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F4bmkrq3)
parent: [Link](https://rebrand.ly/xf21jnm) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fxf21jnm)

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: None

[MBL-14014](https://instructure.atlassian.net/browse/MBL-14014)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.64% | 93.19% | 0.45%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Use Cases/UploadMedia.swift [hunk 2]
```diff
@@ -54,14 +54,17 @@ public class UploadMedia: NSObject, URLSessionDelegate, URLSessionDataDelegate {
...
     func upload() {
         task = env.api.makeRequest(GetMediaServiceRequest()) { (data, _, error) in
-            guard
-                error == nil,
-                let domain = data?.domain.replacingOccurrences(of: "https://", with: ""),
-                let url = URL(string: "https://\(domain)")
-            else {
+            var url: URL?
+            if let domain = data?.domain,
+                domain.starts(with: "http://localhost:") {
```
Comment (bluwave) [inline Core/Core/Use Cases/UploadMedia.swift:59]: Do you want the colon on the end `localhost:` ?  Not sure if this was a typo or you have port too in url

Comment (cobbal) [reply Core/Core/Use Cases/UploadMedia.swift:59]: Yeah, there's a port. Just being paranoid about something like "localhost.notreally.com"

**Pull request merged**